src/cue/cue.cc

```diff
@@ -101,16 +101,27 @@ bool CueLoader::load (const char * cue_filename, VFSFile & file, String & title,
                         base_tuple.set_str (Tuple::Album, s);
                     if ((s = cdtext_get (PTI_GENRE, cdtext)))
                         base_tuple.set_str (Tuple::Genre, s);
+                    if ((s = cdtext_get (PTI_COMPOSER, cdtext)))
+                        base_tuple.set_str (Tuple::Composer, s);
                 }
 
                 Rem * rem = cd_get_rem (cd);
 
                 if (rem)
                 {
                     const char * s;
-                    if ((s = rem_get (REM_DATE, rem)))
-                        base_tuple.set_int (Tuple::Year, str_to_int (s));
+                    if ((s = rem_get (REM_DATE, rem))) {
+                        // We should extract the year out of the string, nothing guarantees it will only be a year.
+                        base_tuple.set_int (Tuple::Year, str_to_int(s));
+                        base_tuple.set_str (Tuple::Date, s);
```

Comment:
How about setting Tuple::Year if the string is all digits (use g_ascii_isdigit() to check), otherwise setting Tuple::Date?